### ISSUE TITLE
Pods 2.8: destroyed conditional logic for pre 2.8 forms on front-end

### ISSUE THREAD

**Issue by JoryHogeveen** [pods-framework/pods#6307]
### Description

PR's https://github.com/pods-framework/pods/pull/5137 and https://github.com/pods-framework/pods/pull/5145 enabled conditional logic for frontend forms.
Since 2.8(.x?), this isn't working anymore.

There are two areas that do not work anymore:
- If you disable DFV for your forms none of the conditional logic is working. No conditional HTML is generated anymore.
- If you enable DFV it's still not working. Fields are disappearing but not showing anymore. (tested on a text based conditional).

### Version

2.8+

### Testing Instructions

Add this in a must-use plugin:

```php
// DFV enabled.
add_action( 'wp_body_open', function () {
	$pod = pods( 'post' );

	$fields = array(
		'post_title' => array( 'label' => 'title', 'type' => 'text', ),
		'website' => array( 'label' => 'website', 'type' => 'text',  ),
		'test' => array( 'label' => 'test', 'type' => 'text', 'depends-on' => array( 'website' => 'yay' ) ),
	);

	echo $pod->form( $fields );
} );
```

```php
// DFV disabled.
add_action( 'wp_body_open', function () {
	$pod = pods( 'post' );

	$fields = array(
		'post_title' => array( 'label' => 'title', 'type' => 'text', 'disable_dfv' => 1 ),
		'website' => array( 'label' => 'website', 'type' => 'text', 'disable_dfv' => 1, 'dependency' => 1 ),
		'test' => array( 'label' => 'test', 'type' => 'text', 'disable_dfv' => 1, 'depends-on' => array( 'website' => 'yay' ) ),
	);

	echo $pod->form( $fields );
} );
```

### Possible Workaround

Not found.

**Comment by sc0ttkclark** [comment]
@zrothauser Something for you here -- the problem essentially is that when you have custom fields defined that are not added directly to the pod configuration itself -- they can't be targeted by field name like we did before (previously stored data-name on the inputs and used that for reference).

If we have a field render on the page that we have a name for, can we add that to the store of field names that can be referenced for dependencies?

**Comment by JoryHogeveen** [comment]
Note: For DFV front-end using the `pods_field_` prefix works.
Example:"

```php
// DFV enabled.
add_action( 'wp_body_open', function () {
	$pod = pods( 'post' );

	$fields = array(
		'post_title' => array( 'label' => 'title', 'type' => 'text', ),
		'website' => array( 'label' => 'website', 'type' => 'text',  ),
		'test' => array( 'label' => 'test', 'type' => 'text', 'depends-on' => array( 'pods_field_website' => 'yay' ) ),
	);

	echo $pod->form( $fields );
} );
```

**Comment by sc0ttkclark** [comment]
Note: Disabling DFV was never set up to support the old conditional logic, it'll go away once we get DFV fully implemented everywhere we disable it right now

**Comment by JoryHogeveen** [comment]
@sc0ttkclark I understand why you'd like to deprecate non-DFV fields but I'm not fond of this idea. React is nice but quite difficult for users to hook into. Regular fields are way more extendible, not only for existing fields but also for new/custom fields.

**Comment by sc0ttkclark** [comment]
This is resolved in work from Pods 3.0 conditional logic which no longer relies on field prefixes on inputs and uses the real field name to reference.
